lib/scrabble.rb

```diff
@@ -15,4 +29,39 @@ def point_values
       "Y"=>4, "Z"=>10
     }
   end
+
+  def score_with_multipliers(word, multipliers, word_multiplier = 1)
+    if !word || !@finder.exists?(word)
+      return 0
+    end
+    letters = word.upcase.chars
+    if letters.length != multipliers.length
+      nil
+    else
+      total = 0
```

Comment:
The pattern of using an outside local variable and having each modify it is a sign that there is a better enumerable suited for the task.